fishtown-analytics/docs.getdbt.com
# create rss feed plugin

## What are you changing in this pull request and why?
This adds a RSS feed for the dbt Cloud Release Notes at: 
https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes

This includes three different RSS formats: `rss`, `atom`, & `json`.

## How this works

This is a custom Docusaurus plugin which runs during the build (locally or on deployment). When the site starts up, the plugin goes through the following steps:

1. Gets all files within the `website/docs/docs/dbt-versions/release-notes directory and all subdirectories.
2. It parses the frontmatter properties for each release note file to build an array of data.
3. A new Feed is initialized with all data in the array using the [feed npm package](https://www.npmjs.com/package/feed).
4. New `rss`, `atom`, and `json` feeds are written to the `static/feeds` directory.

If no new release notes have been added during a build, the three feed files will remain unchanged. 

If a new release note is added, the next build will rewrite the three feed files, and those will be committed with the rest of the updates made for that PR. 

Confirmed the Netlify build updates these feeds during deploy preview (& assuming production) builds with the following steps:

1. Created new [test PR](https://github.com/dbt-labs/docs.getdbt.com/pull/2726) with a new release note added
2. Did not commit the rss feed files under `static/feeds`
3. After the build in the test PR was complete, the preview had the [new release note](https://deploy-preview-2726--docs-getdbt-com.netlify.app/feeds/rss.xml), as it was generated during the Netlify build.

This means we should always have the latest RSS feed data, even if the actual feed files under `static/feeds` aren't committed during an update, as Netlify will generate them at build time.

![image](https://user-images.githubusercontent.com/13647110/213508842-efc1b347-62db-48ab-b1c4-381548faa6f9.png)

## Preview Links
https://deploy-preview-2713--docs-getdbt-com.netlify.app/feeds/rss.xml
https://deploy-preview-2713--docs-getdbt-com.netlify.app/feeds/atom.xml
https://deploy-preview-2713--docs-getdbt-com.netlify.app/feeds/rss.json

## To Test

To test the functionality of the feed, you can click any of the preview links above, which should open a page with the html/json for the feed.

You can also download an RSS Feed reader, such as the Chrome extension [Feeder](https://chrome.google.com/webstore/detail/rss-feed-reader/pnjaodmkngahhkoihejjehlcdlnohgmp?hl=en), and add the rss preview link into the reader.

## TODO
- [x] Set correct links for each item in the RSS feed
- [x] Set date from the frontmatter `tags` array if available
- [x] Open new branch set to merge into this branch. Create a commit within Git, and open new PR to see if RSS feeds are updated that way, or if it has to be done locally. 
- [x] Add logging to the RSS feed plugin which will be visible during Netlify builds
- [x] Before take-live: Update `feedLinks` to use live `siteUrl` variable
